mlir/utils/performance/tuningRunner.py

```diff
@@ -59,42 +60,48 @@ def verifyKernelWithPerfConfig(perfConfig, config, paths: Paths, options: Option
-    # pipe to rocmlir-driver
-    p2 = subprocess.Popen(rocmlirDriverCommand, stdin=p1.stdout, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
-    p1.stdout.close() # Allow p1 to receive a SIGPIPE if p2 exits.
-    # pipe to rocprof + mlir-cpu-runner.
-    p3 = subprocess.Popen(profilerCommand, stdin=p2.stdout, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    p2.stdout.close() # Allow p2 to receive a SIGPIPE if p3 exits.
-    # get output.
-    try:
-        outs, errs = p3.communicate(timeout=600)
-        outs = outs.decode('utf-8')
-        if len(errs) > 0 or not CORRECT_RESULT_RE.search(outs):
-            print(f"""Verification failed:
+        print(rocmlirGenCommand, file=sys.stderr)
+
+    prevdir = os.getcwd()
+    with tempfile.TemporaryDirectory() as tmpdir:
+        try:
+            os.chdir(tmpdir)
+            # invoke rocmlir-gen.
+            p1 = subprocess.Popen(rocmlirGenCommand.split(), stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
+            # pipe to rocmlir-driver
+            p2 = subprocess.Popen(rocmlirDriverCommand, stdin=p1.stdout, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
+            p1.stdout.close() # Allow p1 to receive a SIGPIPE if p2 exits.
+            # pipe to rocprof + mlir-cpu-runner.
+            p3 = subprocess.Popen(profilerCommand, stdin=p2.stdout, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            p2.stdout.close() # Allow p2 to receive a SIGPIPE if p3 exits.
+            # get output.
+            try:
+                outs, errs = p3.communicate(timeout=600)
+                outs = outs.decode('utf-8')
+                if len(errs) > 0 or not CORRECT_RESULT_RE.search(outs):
+                    print(f"""Verification failed:
 Output = {outs}
-Errors = {errs.decode('utf-8')}""")
-            return np.nan
-    except subprocess.TimeoutExpired:
-        print("Verification timed out")
-        p3.kill()
-        outs, errs = p3.communicate()
-        return np.nan
-    nanoSeconds = perfRunner.getNanoSeconds(perfRunner.BENCHMARKING_RESULT_FILE_NAME)
+Errors = {errs.decode('utf-8')}""", file=sys.stderr)
+                    return np.nan
+            except subprocess.TimeoutExpired:
+                print("Verification timed out", file=sys.stderr)
+                p3.kill()
+                outs, errs = p3.communicate()
+                return np.nan
+            nanoSeconds = perfRunner.getNanoSeconds(perfRunner.BENCHMARKING_RESULT_FILE_NAME)
+        finally:
+            os.chdir(prevdir)
     return nanoSeconds
 
 def getWinningConfig(tuningOutput, config, allData, options: Options):
     maxTFlops = -np.inf
     winningConfig = "None"
     for i, result in enumerate(tuningOutput):
         result = result.decode('utf-8').strip()
-        if i > 0 and i % 100 == 0:
-            print(f"Tested {i} configs, best perf {maxTFlops} TFlops on perf_config {winningConfig}")
+        if not options.quiet and i > 0 and i % 100 == 0:
```

Comment:
One note: Jenkins relies on scripts printing out to stderr/stdout regularly to enforce timeouts - are you sure you're not breaking this here?